searchcore/src/vespa/searchcore/proton/server/bucketmovejobv2.cpp

```diff
@@ -260,6 +267,13 @@ BucketMoveJobV2::considerBucket(const bucketdb::Guard & guard, BucketId bucket)
     } else {
         _buckets2Move.erase(bucket);
         cancelMovesForBucket(bucket);
+        if (_movers2Complete.find(bucket) != _movers2Complete.end()) {
+            // This will continue until all inflight operations on this bucket is completed
+            _master.execute(makeLambdaTask([this, bucket]() {
```

Comment:
This looks like the wrong place to schedule the extra check. If a bucket executor has a task for the bucket but waits for the bucket lock then you get a busy loop scheduling considerBucket tasks. Similarly if we end up with a stray leftover _moversToComplete entry.

Reply:
I agree this is not a good solution.